_projects/optimize_caching.md

```diff
@@ -62,23 +67,24 @@ increase performance even further.
 
 ##### CODING PHASE 1
 
-* The Proposed change to not explore unchanged directories has been  
-  implemented.
-* Thorough tests have been performed to concretely ascertain a measurable  
-  performance gain.
+* `FileFactory` class has been implemented and the file contents in the
+  file-dict have been replaced with the factory objects.
+* `Directory` class has been implemented.
 * The corresponding code is properly tested, documented, reviewed, and merged.
 
 ##### CODING PHASE 2
 
-* Section-based caching has been implemented.
+* The Proposed change to ignore unchanged directories has been implemented.
+* Thorough tests have been performed to concretely ascertain a measurable  
+  performance gain
+* At least two of the cache control flags have been implemented.
 * The corresponding code has been thoroughly tested, documented, reviewed, and  
   merged.
-* At least one of the proposed performance bottleneck improvements have been  
-  implemented, tested, documented, and merged.
 
 ##### CODING PHASE 3
 
-* At least an additional proposed performance bottleneck improvement has been  
-  implemented, tested, documented, and merged.
+* All the remaining cache control flags like `--cache-optimize`,
```

Comment:
depends on cEP being accepted.

Reply:
@jayvdb the cEP has been merged https://github.com/coala/cEPs/pull/146/

Can we merge this too?

Reply:
Even without cEP acceptance, those cache-control flags are needed in any way.

Reply:
Maybe not specific ones like `--cache-optimize`, but `--cache-compression` for example is for nearly every cache implementation viable.